File: Java/Matrizes/soma_matrizes.java
// Exercício: Soma Matrizes
// Objetivo: Ler duas matrizes A e B de mesma dimensão (M x N), gerar a matriz C contendo a soma dos elementos correspondentes e exibi-la.

import java.util.Scanner;

public class soma_matrizes {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        
        int n, m;

        System.out.print("Quantas linhas vai ter cada matriz? ");
        m = sc.nextInt();
        System.out.print("Quantas colunas vai ter cada matriz? ");
        n = sc.nextInt();

        int[][] matA = new int[m][n];
        int[][] matB = new int[m][n];
        int[][] matC = new int[m][n];

        System.out.println("Digite os valores da matriz A: ");
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                matA[i][j] = sc.nextInt();
            }
        }

        System.out.println("Digite os valores da matriz B: ");
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                matB[i][j] = sc.nextInt();
            }
        }

        System.out.println("MATRIZ SOMA: ");

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                matC[i][j] = matA[i][j] + matB[i][j];
            }
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print(matC[i][j] + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

File: Python/matrizes/soma_matrizes.py
# Exercício: Soma Matrizes
# Objetivo: Ler duas matrizes M x N (A e B), gerar uma terceira matriz C que seja a soma de A e B, e exibi-la.

m = int(input("Quantas linhas vai ter cada matriz? "))
n = int(input("Quantas colunas vai ter cada matriz? "))

matA = []
matB = []
matC = []


print("Digite os valores da matriz A: ")
for i in range(m):
    linha = []

    for j in range(n):
        coluna = int(input(f"Elemento [{i},{j}]: "))
        linha.append(coluna)
    matA.append(linha)

print("Digite os valores da matriz B: ")
for i in range(m):
    linha = []

    for j in range(n):
        coluna = int(input(f"Elemento [{i},{j}]: "))
        linha.append(coluna)
    matB.append(linha)

print("MATRIZ SOMA: ")
for i in range(m):
    linha = []

    for j in range(n):
        soma = matA[i][j] + matB[i][j]
        linha.append(soma)
    matC.append(linha)

for i in range(m):
    for j in range(n):
        print(f"{matC[i][j]}", end=" ")
    print()